### PR TITLE
histogram: impl Drop for HistogramTimer

### PR DESCRIPTION
**Pull request by overvenus**
Drop as observe duration.

### CODE REVIEW

File: examples/example_hyper.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ fn main() {
         .unwrap()
         .handle(move |_: Request, mut res: Response| {
             HTTP_COUNTER.inc();
-            let timer = HTTP_REQ_HISTOGRAM.start_timer();
+            let mut timer = HTTP_REQ_HISTOGRAM.start_timer();
```
Comment (siddontang) [inline examples/example_hyper.rs]: can we use `let _timer =` instead?

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @disksing

File: src/histogram.rs [hunk 2]
```diff
@@ -194,25 +194,39 @@ impl Default for HistogramCore {
     }
 }
 
-/// `HistogramTimer` represents an event being timed.
+/// `HistogramTimer` represents an event being timed. When the timer goes out of
+/// scope, the duration will be observed, or call `observe_duration` to manually
+/// observe.
```
Comment (disksing) [inline src/histogram.rs]: Add some notes to remind that a Timer could be observed only once (automatically or manually).

**Comment by disksing** [comment]
LGTM

**Pull request merged**